python/tank/pipelineconfig_factory.py

```diff
@@ -15,22 +15,46 @@
 import cPickle as pickle
 
 from .errors import TankError
-from .platform import constants 
+from .platform import constants
 from .util import shotgun
 from . import pipelineconfig_utils
 from .pipelineconfig import PipelineConfiguration
 
 
+def _massage_path(path):
+    """
+    Converts all slashes to forward slashes and normalizes it.
+
+    :param path: Path to massage.
+
+    :returns: A normalized path with only forward slashes
+    """
+    return os.path.normpath(path.replace("\\", "/"))
+
+
+def _is_path_in_paths(path, paths):
```

Comment:
Is this the wrong approach? Instead of adding methods for "fuzzy" comparison, instead make sure to normalize all inputs into the system first, then run all logic under the assumption that things are normalized. In my experience, this approach can easily get messy quickly. Instead, just normalize everything up front. Does that make sense? 

You use the method like this:

```
if not _is_path_in_paths(config_context_path, local_pc_paths):
```

I suggest instead doing the follwoing:

```
normalized_config_context_path = _normalize_path(config_context_path)
if not normalized_config_context_path in local_pc_paths:
```